implementations/apply.js

```diff
@@ -0,0 +1,40 @@
+/*
+  The apply() method calls a function with a given this value, and arguments provided as an array (or an array-like object).
+
+
+  MDN Link: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply
+
+
+  Parameters
+
+  * otherThis
+    The value of this provided for the call to func. 
+  * arr
+    Optional. An array-like object, specifying the arguments with which func should be called, or null or undefined if no arguments should be provided to the function. 
+
+  Return value
+    The result of calling the function with the specified this value and arguments.
+*/
+
+Function.prototype.myApply = function (otherThis, arr) {
+  otherThis = otherThis || global;
+  var uniqueID = "00" + Math.random();
+  while (otherThis.hasOwnProperty(uniqueID)) {
+    uniqueID = "00" + Math.random();
+  }
+  otherThis[uniqueID] = this;
```

Comment:
Can you use this line of code?
`otherThis.__proto__.uniqueID = this;`

Reply:
@TheSTL Why do we need to use this instead of `otherThis[uniqueID] = this;`  . Can you elaborate ?

Reply:
By doing this we are making a pair of key-value in `otherThis` where key is some random id and value is function on which apply is called. When we call the function of `otherThis` (`otherThis.uniqueID(...arr)`) then in function `this` value will be `otherThis`.

Reply:
@TheSTL  The same is being done by @tamirisapbonicenha in line no 30 & 35 
`result = otherThis[uniqueID]();`.

The only benefit what I can see from using `otherThis.__proto__.uniqueID = this;` in place of 
`otherThis[uniqueID] = this;` is that there is no chance of collision of the unique id in the prorotype object.
Moreover is __proto__ is a browser implementation , instead of a built in js implementation

Reply:
Actually problem is  the uniqueId property is printed .You can check snapshot https://github.com/knaxus/native-javascript/pull/23#pullrequestreview-298972630

Reply:
Thanks @ashishdutta007 for telling me `Moreover is __proto__ is a browser implementation , instead of a built in js implementation.`. I'm testing things on browser .